### PR TITLE
[AKS] support custom private dns zone for aks private cluster

### PR DESCRIPTION
**Pull request by levimm**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
aks

**Comment by levimm** [comment]
@fengzhou-msft  The tests failed because of "ERROR: Requested azure-cli from https://azurecliprod.blob.core.windows.net/edge/azure-cli/azure_cli-2.16.0.post1-py3-none-any.whl has different version in metadata: '2.16.0'"
Do you know how to fix that?

**Review by feiskyer — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -248,7 +248,7 @@
         - name: --private-dns-zone
           type: string
           short-summary: (PREVIEW) private dns zone mode for private cluster.
-          long-summary: Select between system and none. If not set, defaults to type system. Requires --enable-private-cluster to be used.
+          long-summary: Allowed values are system, none or your custom private dns zone resource id. If not set, defaults to type system. Requires --enable-private-cluster to be used.
```
Comment (gossion) [inline src/aks-preview/azext_aks_preview/_help.py:251]: nit: better to add quote to "system", "none"

**Review by gossion — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/_help.py (hunk 1 above)
Comment (levimm) [reply src/aks-preview/azext_aks_preview/_help.py]: quote added. thanks

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**